src/late_multicellular_stage/DualContourer.cs

```diff
@@ -0,0 +1,423 @@
+
+            // Parallel to x face, plane going through z = 1 and y = 1
+            if (cube.Points[0, 1, 1] ^ cube.Points[1, 1, 1])
+            {
+                tris.Add(new Vector3I(0, 0, 0));
+                tris.Add(new Vector3I(0, 1, 1));
+                tris.Add(new Vector3I(0, 0, 1));
+
+                tris.Add(new Vector3I(0, 0, 0));
+                tris.Add(new Vector3I(0, 1, 0));
+                tris.Add(new Vector3I(0, 1, 1));
+
+                if (cube.Points[0, 1, 1])
+                {
+                    int lastID = tris.Count - 1;
+
+                    (tris[lastID], tris[lastID - 1]) = (tris[lastID - 1], tris[lastID]);
+                    (tris[lastID - 3], tris[lastID - 4]) = (tris[lastID - 4], tris[lastID - 3]);
+                }
+            }
+
+            // Parallel to z face, plane going through x = 1 and y = 1
+            if (cube.Points[1, 1, 0] ^ cube.Points[1, 1, 1])
+            {
+                tris.Add(new Vector3I(0, 0, 0));
+                tris.Add(new Vector3I(1, 0, 0));
+                tris.Add(new Vector3I(1, 1, 0));
+
+                tris.Add(new Vector3I(0, 0, 0));
+                tris.Add(new Vector3I(1, 1, 0));
+                tris.Add(new Vector3I(0, 1, 0));
+
+                if (cube.Points[1, 1, 0])
+                {
+                    int lastID = tris.Count - 1;
+
+                    (tris[lastID], tris[lastID - 1]) = (tris[lastID - 1], tris[lastID]);
+                    (tris[lastID - 3], tris[lastID - 4]) = (tris[lastID - 4], tris[lastID - 3]);
+                }
+            }
+
+            // Parallel to y face, plane going through z = 1 and x = 1
+            if (cube.Points[1, 0, 1] ^ cube.Points[1, 1, 1])
+            {
+                tris.Add(new Vector3I(0, 0, 0));
+                tris.Add(new Vector3I(1, 0, 1));
+                tris.Add(new Vector3I(1, 0, 0));
+
+                tris.Add(new Vector3I(0, 0, 0));
+                tris.Add(new Vector3I(0, 0, 1));
+                tris.Add(new Vector3I(1, 0, 1));
+
+                if (cube.Points[1, 0, 1])
+                {
+                    int lastID = tris.Count - 1;
+
+                    (tris[lastID], tris[lastID - 1]) = (tris[lastID - 1], tris[lastID]);
+                    (tris[lastID - 3], tris[lastID - 4]) = (tris[lastID - 4], tris[lastID - 3]);
+                }
+            }
+
+            if (reverseTriangleFaces)
+            {
+                for (int j = 1; j < tris.Count; j += 3)
+                {
+                    (tris[j], tris[j + 1]) = (tris[j + 1], tris[j]);
+                }
+            }
+
+            lookupTableInt.Add(i, tris.ToArray());
+        }
+    }
+
+    public bool IsInShape(Vector3 pos)
+    {
+        if (MathFunction!.GetValue(pos) > MathFunction.SurfaceValue)
+            return true;
+
+        return false;
+    }
+
+    public Mesh DualContour()
+    {
+        // GD.Print("Starting mesh generation");
+
+        var sw = new Stopwatch();
```

Comment:
This stopwatch seems like leftover debug code, unless you want to keep it here printing the times, which I think would be fine for now to kind of track how fast this code is.

Reply:
Added a single debug print for overall time spent.